### PR TITLE
Migrate base_import_async to 9.0

### PR DESCRIPTION
**Pull request by sbidoul**
including latests commit from the 8.0 branch

TODO:
- [x] fix problem with qweb and/or js that prevents the view to load

### CODE REVIEW

**Comment by sbidoul** [comment]
Tested with Connector 9.0.1.0.2

**Comment by max3903** [comment]
:+1: Code review, no test

**Comment by lmignon** [comment]
:+1: Code review + test

File: base_import_async/models/base_import_async.py [hunk 1]
```diff
@@ -61,8 +61,8 @@ def _create_csv_attachment(session, fields, data, options, file_name):
     # write csv
     f = StringIO()
     writer = csv.writer(f,
-                        delimiter=options.get(OPT_SEPARATOR),
-                        quotechar=options.get(OPT_QUOTING))
+                        delimiter=str(options.get(OPT_SEPARATOR)),
+                        quotechar=str(options.get(OPT_QUOTING)))
```
Comment (guewen) [inline base_import_async/models/base_import_async.py:65]: Couldn'it end up with `'None'` as delimiter and quotechar if no option is set?

Comment (sbidoul) [inline base_import_async/models/base_import_async.py:65]: @guewen could be. It's not been an issue in practice so far. I guess it would impact only people using the API, which is kind of undocumented anyway.

Comment (sbidoul) [inline base_import_async/models/base_import_async.py:65]: @guewen note this change is part of the 8.0 forward port. So if that needs changing I'd do it in another PR.

Comment (guewen) [inline base_import_async/models/base_import_async.py:65]: I see, thanks for clarifying

**Comment by guewen** [comment]
:+1:

**Comment by atchuthan** [comment]
:+1: Code review
